From b207b1286a00e36a97900387ff11e58c973f68f7 Mon Sep 17 00:00:00 2001
From: Pavan Nikhilesh <pbhagavatula@marvell.com>
Date: Tue, 2 Jul 2019 16:41:11 +0530
Subject: [PATCH 0231/1921] octeontx2-af: fix TIM slot to lf lookup

When we retrieve slot to LF mapping we have to read all the values
enumerated by TIM_PRIV_LF(0..255)_CFG rather than the first LFs slot
mapping.

Change-Id: I7b71267b3b535bb8933fd23db52724b7294b36d0
Signed-off-by: Pavan Nikhilesh <pbhagavatula@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/12467
Reviewed-by: Stanislaw Kardach <Stanislaw.Kardach@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c
index d0ea06f0e23a..3f07334c49b2 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_tim.c
@@ -69,7 +69,8 @@ int rvu_lf_lookup_tim_errata(struct rvu *rvu, struct rvu_block *block,
 		return TIM_AF_LF_INVALID;
 
 	for (i = 0; i < block->lf.max; i++) {
-		val = rvu_read64(rvu, block->addr, block->lfcfg_reg);
+		val = rvu_read64(rvu, block->addr, block->lfcfg_reg |
+				 (i << block->lfshift));
 		if ((((val >> 8) & 0xffff) == pcifunc) &&
 				(val & 0xff) == slot)
 			return i;
-- 
2.31.1

